frontend/model/contracts/misc/flowTyper.js

```diff
@@ -341,6 +341,18 @@ export const string = (
   : TypeValidator<string>
 )
 
+export const stringMax = (numChar: number, key: string = ''): TypeValidator<string> => {
+  if (!isNumber(numChar)) { throw new Error('param for stringMax must be number') }
+
+  return function stringMax (value, _scope = '') {
+    string(value, _scope)
+    if (value.length <= numChar) return value
+    throw new TypeError(
+      key ? `[stringMax] string type '${key}' cannot exceed ${numChar} characters` : `[stringMax] cannot exceed ${numChar} characters`
+    )
+  }
+}
+
```

Comment:
Let me know if this can be simplified and improved.

Reply:
@SebinSong you did a really job here. It's almost perfect. I worked with Claude to see if some improvements could be made, and here's what "we" came up with:

```js
export const stringMax = (numChar: number, key: string = ''): TypeValidator<string> => {
  if (!isNumber(numChar)) {
    throw new Error('param for stringMax must be number')
  }

  function stringMax (value, _scope = '') {
    const strValue = string(value, _scope)
    if (strValue.length <= numChar) return strValue
    throw validatorError(
      stringMax,
      value,
      _scope,
      key
        ? `string type '${key}' cannot exceed ${numChar} characters`
        : `cannot exceed ${numChar} characters`
    )
  }
  stringMax.type = () => `string(max: ${numChar})`
  return stringMax
}
```

This version is very similar to yours, but it retains greater consistency with the rest of the validation functions by reusing the return value of `string` validation function, using `validationError` instead of `TypeError`, and defining a `type`.